event_test.go

```diff
@@ -0,0 +1,55 @@
+package stripe
+
+import (
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+)
+
+func TestGetObjValue(t *testing.T) {
+	event := &Event{
+		Data: &EventData{
+			Obj: map[string]interface{}{
+				"top_level_key": "top_level",
+				"integer_key":   123,
+				"map": map[string]interface{}{
+					"nested_key": "nested",
+				},
+				"slice": []interface{}{
+					"index-0",
+					"index-1",
+					"index-2",
+				},
+				"slice_of_maps": []interface{}{
+					map[string]interface{}{
+						"slice_nested_key": "slice_nested",
+					},
+				},
+			},
+		},
+	}
+
+	assert.Equal(t, "top_level", event.GetObjValue("top_level_key"))
+
+	// Check that it coerces non-string values into strings (this behavior is
+	// somewhat questionable, but I'm going with how it already works)
+	assert.Equal(t, "123", event.GetObjValue("integer_key"))
+
+	assert.Equal(t, "nested", event.GetObjValue("map", "nested_key"))
+	assert.Equal(t, "index-1", event.GetObjValue("slice", "1"))
+	assert.Equal(t, "slice_nested",
+		event.GetObjValue("slice_of_maps", "0", "slice_nested_key"))
+
+	// By design a `nil` just returns an empty string
+	assert.Equal(t, "", event.GetObjValue("bad_key"))
+
+	// Panic conditions. Usually the function tries to just return a value is
+	// fairly forgiving, but it does panic under certain obviously impossible
+	// cases.
+	assert.Panicsf(t, func() {
+		event.GetObjValue("slice", "string_key")
+	}, "Cannot access nested slice element with non-integer key: %s", "bad_key")
```

Comment:
I'm likely missing something since the tests are passing, but shouldn't the assert expect `"string_key"`, not `"bad_key"`?